jakarta-eclipse/org.eclipse.lsp4jakarta.core/src/org/jakarta/jdt/beanvalidation/BeanValidationQuickFix.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+* Copyright (c) 2020 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation - initial API and implementation
+*******************************************************************************/
+package org.jakarta.jdt.beanvalidation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.IBinding;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.internal.corext.dom.Bindings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.jakarta.codeAction.IJavaCodeActionParticipant;
+import org.jakarta.codeAction.JavaCodeActionContext;
+import org.jakarta.codeAction.proposal.AddConstructorProposal;
+import org.jakarta.codeAction.proposal.ChangeCorrectionProposal;
+import org.jakarta.codeAction.proposal.DeleteAnnotationProposal;
+import org.jakarta.codeAction.proposal.ModifyModifiersProposal;
+import org.jakarta.codeAction.proposal.quickfix.RemoveAnnotationConflictQuickFix;
+
+/**
+ * Quickfix for fixing {@link BeanValidationConstants#DIAGNOSTIC_CODE_Static} error by either action 
+ * 1. Removing constraint annotation on static field or method
+ * 2. Removing static modifier from field or method
+ * 
+ * @author Leslie Dawson (lamminade)
+ *
+ */
+public class BeanValidationQuickFix implements IJavaCodeActionParticipant {
+
+    @Override
+    public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+            IProgressMonitor monitor) throws CoreException {
+        ASTNode node = context.getCoveredNode();
+        IBinding parentType = getBinding(node);
+
+        List<CodeAction> codeActions = new ArrayList<>();
+        codeActions.add(removeConstraintAnnotations(diagnostic, context, parentType));
+
+        if (diagnostic.getCode().getLeft().equals(BeanValidationConstants.DIAGNOSTIC_CODE_STATIC)) {
+            codeActions.add(removeStaticModifier(diagnostic, context, parentType));
+        }
```

Comment:
When both static modifier and incorrect annotation is on the field, there is a duplicate code action:
![image](https://user-images.githubusercontent.com/26146482/110966368-2ef78700-8323-11eb-83b9-27b67f8e8565.png)

Reply:
Option to remove static modifier does not seem to appear on methods:
![image](https://user-images.githubusercontent.com/26146482/110966462-433b8400-8323-11eb-805d-64b6a053a734.png)

Reply:
Consider adding the above scenarios to tests as well

Reply:
dcccd8d